cmd/shell.go

```diff
@@ -263,6 +267,41 @@ type Queue struct {
 	Server string
 }
 
+// RateLimitConfig contains all application layer rate limiting policies
+type RateLimitConfig struct {
+	TotalCertificates RateLimitPolicy `yaml:"totalCertificates"`
+}
+
+// RateLimitPolicy describes a generate limiting policy
+type RateLimitPolicy struct {
+	Window       time.Duration    `yaml:"-"`
+	WindowString string           `yaml:"window"`
+	Threshold    int64            `yaml:"threshold"`
+	Overrides    map[string]int64 `yaml:"overrides"`
+}
+
+// LoadRateLimitPolicies loads various rate limiting policies from a YAML
+// configuration file
+func LoadRateLimitPolicies(filename string) (RateLimitConfig, error) {
+	contents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return RateLimitConfig{}, err
+	}
+	var rlc RateLimitConfig
+	err = yaml.Unmarshal(contents, &rlc)
+	if err != nil {
+		return RateLimitConfig{}, err
+	}
+
+	totalCertWindow, err := time.ParseDuration(rlc.TotalCertificates.WindowString)
```

Comment:
Oh. Yeah, we can just make it a ConfigDuration and teach it `UnmarshalYAML(unmarshal func(interface{}) error)`, as well.